torch/csrc/jit/codegen/cuda/scheduler/transpose.cpp

```diff
@@ -432,10 +556,65 @@ void scheduleTranspose(Fusion* fusion, const TransposeParams& params) {
   auto inner_most_id1 = scheduler_utils::innerMostRootDim(reference1);
   auto inner_most_id2 = scheduler_utils::innerMostRootDim(reference2);
 
-  auto inner_most_pos1_in_ref1 =
+  //////////////////////////////////////////
+  // Step 1: Make virtual inner most dims //
+  //////////////////////////////////////////
+
+  // See note [Supporting small transpose dimensions]
+
+  // split big dims so that we have enough dimensions available to merge with
+  // inner-most dims to create the virtual inner-most dim
+  scheduler_utils::splitDims(reference1, params.split_before_tiling);
+  auto merged1 = scheduler_utils::mergeDims(
```

Comment:
Nit comment: Merging reference 1 dims_merged_with_1 but updating dims_merged_with_2 based on the changes in the dimensions that were merged. So we can then run merge with dims_merged_with_2.

Reply:
added